src/copyright/ui/ajax-copyright-hist.php

```diff
@@ -100,7 +100,15 @@ function Output()
     else if($action=="update" || $action=="delete" || $action=="undo")
     {
       $id = GetParm("id", PARM_STRING);
-      list($upload, $item, $hash, $type) = explode(",", $id);
+      $getEachID = array_filter(explode(",", trim($id, ',')));
+      if(count($getEachID) == 4) {
+        list($upload, $item, $hash, $type) = $getEachID;
+      } else {
+        return new JsonResponse(array(
+                 "error" => 'bad request'
+              ), JsonResponse::HTTP_INTERNAL_SERVER_ERROR
```

Comment:
Please return `HTTP_BAD_REQUEST` to match the error message.
```suggestion
              ), JsonResponse::HTTP_BAD_REQUEST
```